projects/components/src/table/table.service.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable, Injector } from '@angular/core';
+import { FilterAttribute } from '../filter-bar/filter-attribute';
+import { FilterType } from '../filter-bar/filter-type';
+import { TableCellParserLookupService } from './cells/table-cell-parser-lookup.service';
+import { TableCellRendererLookupService } from './cells/table-cell-renderer-lookup.service';
+import { CoreTableCellRendererType } from './cells/types/core-table-cell-renderer-type';
+import { TableCdkDataSource } from './data/table-cdk-data-source';
+import { TableColumnConfig, TableColumnConfigExtended } from './table-api';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TableService {
+  private static readonly STATE_ATTRIBUTE: FilterAttribute = {
+    name: '$$state',
+    displayName: 'Row State',
+    type: '$$state' as FilterType
+  };
+
+  public constructor(
+    private readonly injector: Injector,
```

Comment:
Since this is a root service, it will get a root injector rather than the injector in the context of the current table.

Reply:
Correct. Root injector is fine. I renamed to clear up any assumptions.

Reply:
Feels like it's making an assumption on the parsers, although I agree that for all practical applications it should be the same.